src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -195,6 +197,15 @@ var ReactCompositeComponentMixin = {
           Component.displayName || Component.name || 'Component'
         );
       }
+
+      propsMutated = !shallowEqual(inst.props, publicProps);
+      if (propsMutated) {
+        warning(
+          propsMutated,
+          'Mutated props not allowed in %s(...)',
```

Comment:
If someone is debugging, just saying "Mutated props not allowed in..." doesn't really describe what's wrong.  Someone who is writing a moderately complex component might not see what the issue is.

I would say something like "Constructor (of %s) may not invoke `super(...)` with props that differ from the props that were initially passed in from the component's owner."